src/sentry/static/sentry/app/views/settings/projectSourceMaps/list/index.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import {RouteComponentProps} from 'react-router/lib/Router';
+import styled from '@emotion/styled';
+
+import {t} from 'app/locale';
+import AsyncView from 'app/views/asyncView';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import {Organization, Project, SourceMapsArchive} from 'app/types';
+import routeTitleGen from 'app/utils/routeTitle';
+import TextBlock from 'app/views/settings/components/text/textBlock';
+import SearchBar from 'app/components/searchBar';
+import Pagination from 'app/components/pagination';
+import {PanelTable} from 'app/components/panels';
+import space from 'app/styles/space';
+
+import SourceMapsArchiveRow from './sourceMapsArchiveRow';
+
+type Props = RouteComponentProps<{orgId: string; projectId: string}, {}> & {
+  organization: Organization;
+  project: Project;
+};
+
+type State = AsyncView['state'] & {
+  archives: SourceMapsArchive[];
+};
+
+class ProjectSourceMaps extends AsyncView<Props, State> {
+  getTitle() {
+    const {projectId} = this.props.params;
+
+    return routeTitleGen(t('Source Maps'), projectId, false);
+  }
+
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      archives: [],
+    };
+  }
+
+  getEndpoints() {
+    const {params, location} = this.props;
+    const {orgId, projectId} = params;
+
+    const endpoints: ReturnType<AsyncView['getEndpoints']> = [
+      [
+        'archives',
+        `/projects/${orgId}/${projectId}/files/source-maps/`,
+        {query: {query: location.query.query}},
+      ],
+    ];
+
+    return endpoints;
+  }
+
+  handleSearch = (query: string) => {
+    const {location, router} = this.props;
+
+    router.push({
+      ...location,
+      query: {...location.query, cursor: undefined, query},
+    });
+  };
+
+  getQuery() {
+    const {query} = this.props.location.query;
+
+    return typeof query === 'string' ? query : undefined;
```

Comment:
You could use `app/utils/queryString:decodeScalar()` to do this conversion/extraction from the query string.

Reply:
Nice TIL!